### DISCUSSION TITLE
Getting "Module not found" when renaming path in `tsconfig.json`

### DISCUSSION THREAD

**Discussion by macintacos** [vercel/next.js#32141] [Help]
### What version of Next.js are you using?

It says "latest" in package.json

### What version of Node.js are you using?

17.0.1

### What browser are you using?

Firefox

### What operating system are you using?

macOS 12.0.1

### How are you deploying your application?

Locally with `npm run dev`

### Describe the Bug

I'm learning Next.js for the first time, and I'm walking through the [tutorial](https://nextjs.org/learn/basics/create-nextjs-app). I got to [the section about layouts](https://nextjs.org/learn/basics/assets-metadata-css/layout-component) and wanted to change the relative import to one that was a little more "pretty", so I searched the documentation and found this page which does exactly what I wanted: [Absolute Imports and Module path aliases](https://nextjs.org/docs/advanced-features/module-path-aliases)

However it doesn't seem to work. VSCode is able to resolve the import fine (I get auto-completion and everything) but the Next.js compiler breaks telling me that the module can't be found. It works fine with the relative import.

Here's the full error output in the terminal:

```
error - ./pages/posts/first-post.tsx:2:0
Module not found: Can't resolve '@/components/layout'
  1 | import Head from "next/head";
> 2 | import Layout from "@/components/layout";
  3 | import Link from "next/link";
  4 | 
  5 | export default function FirstPost() {

https://nextjs.org/docs/messages/module-not-found
```

### Expected Behavior

That the page would render. It does not. This would be the "first-post" route that you've been building during the tutorial.

### To Reproduce

Run `npx create-next-app nextjs-blog --use-npm --example "https://github.com/vercel/next-learn/tree/master/basics/learn-starter"`.

Run through the tutorial, but change everything to `.tsx` files instead of `.js`. Next does a fantastic job of telling you what to do there to get properly set up when running `npm run dev`, so just follow those instructions :)

Copy the configuration below and then get `npm run dev` to run the app, and then navigate to `posts/first-post`. Compiler error!

Here's my `tsconfig.json` that you can use to reproduce:

```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "incremental": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "paths": {
      "@/components/*": ["./components/*"]
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"]
}
```

And here's what's in `pages/posts/first-post.tsx`:

```tsx
import Head from "next/head";
import Layout from "@/components/layout";
import Link from "next/link";

export default function FirstPost() {
  return (
    <Layout>
      <Head>
        <title>First Post</title>
      </Head>

      <h1>First Post</h1>
      <h2>
        <Link href="/">
          <a>Back to home</a>
        </Link>
      </h2>
    </Layout>
  );
}
```

**Comment by balazsorban44** [comment]
Could you try adding `"baseUrl": ".",`  to your `tsconfig.json > compilerOptions`.

https://www.typescriptlang.org/tsconfig/#baseUrl

**Reply by rolaru** [reply]
Thanks! This worked perfectly for me, though I don't understand why it didn't work with "baseUrl": "/" 🤔
